packages/core/docs/development/public-api.stories.mdx

```diff
@@ -46,7 +46,36 @@ set the property.
 Lit-element will automatically sync attributes and properties together unless
 specified otherwise. While attributes are always string values in HTML, lit-element
 and the `@property` decorator will attempt to convert the attribute value based
-on the value type that was set on the `@property` decorator. [This includes shallow objects and arrays.](https://lit-element.polymer-project.org/guide/properties#initialize-markup)
+on the value type that was set on the `@property` decorator.
+
+### Primititve Values Only
+
+While lit-element keeps both properties and attributes in sync we do not use
+complex object types like `object` and `array` on the API. Since HTML cannot represent
+JavaScript objects, lit-element must JSON parse attributes and reflect them anytime
+there is a change to ensure the JavaSCript property and HTML attribute are in sync.
```

Comment:
Should we make a note about making an exception to this rule for i18n?

Reply:
Oh I see the note below. Premature comment. :)

Reply:
Good point, will update this

Reply:
Ah lol I see that now as well 🥇